src/ray/raylet/scheduling_resources.h

```diff
@@ -227,6 +271,14 @@ class ResourceIds {
   /// A vector of pairs of resource ID and a fraction of that ID (the fraction
   /// is at least zero and strictly less than 1).
   std::vector<std::pair<int64_t, double>> fractional_ids_;
+  /// int to track the greatest id of the resource - this needs to be maintained to
+  /// ensure unique ids are used when creating new resources. This keeps incrementing,
+  /// and resources may have gaps in between.
```

Comment:
I don't think it's acceptable to have gaps in the IDs. That will give the user an inconsistent experience (e.g., what IDs they get will depend on when things get scheduled). Instead, how about we simply don't provide IDs for dynamic resources?

Reply:
I prefer not providing IDs as well.